### PR TITLE
add playground for toolbarMenu

### PR DESCRIPTION
**Pull request by ektasawant**
Fixes 3529 .

#### Changes proposed in this Pull Request:
- Playground for ToolbarMenu
- 
- 

#### To Test:
- /components/toolbar-menu/playground

### CODE REVIEW

**Comment by jeffvg** [comment]
looks good. Can we make toolbar Menu Items clickable to dismiss toolbar menu

File: docs/src/componentDocs/ToolbarMenu/playground/PreviewComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    const getIconWithProp = (icon: string, iconProps: any): any => {
+        const index = icon.lastIndexOf('/>');
+        const result = icon.slice(0, index) + iterateIconProps(iconProps) + icon.slice(index);
+        return result;
+    };
+
+    const generateCodeSnippet = (): string => {
+        const jsx = `<ToolbarMenu
+    icon={${getIconWithProp(toolbarMenuProps.icon, { htmlColor: `${Colors.green[500]}` })}}
```
Comment (huayunh) [inline docs/src/componentDocs/ToolbarMenu/playground/PreviewComponent.tsx:32]: you can remove this htmlColor thing... should not be the main focus of this code snippet

File: docs/src/componentDocs/ToolbarMenu/playground/ToolbarMenuConfig.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { ComponentType } from '../../../__types__';
+
+export const toolbarMenuConfig: ComponentType = {
+    componentName: 'Toolbar Menu',
+    props: [
+        {
+            propName: 'icon',
+            inputType: 'select',
+            inputValue: '<GradeA />',
+            options: ['<GradeA />', '<GradeB />'],
```
Comment (huayunh) [inline docs/src/componentDocs/ToolbarMenu/playground/ToolbarMenuConfig.tsx:10]: I will nominate these options to be [`<Place />`](https://fonts.google.com/icons?selected=Material%20Icons%3Aplace%3A), [`<Home />`](https://fonts.google.com/icons?icon.set=Material+Icons&icon.style=Filled&icon.query=home) and `undefined` instead

File: docs/src/componentDocs/ToolbarMenu/playground/PreviewComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            str = `${str}` + `${prop}="${iconProps[prop]}" `;
+        }
+        return str;
+    };
+
+    const getIconWithProp = (icon: string, iconProps: any): any => {
+        const index = icon.lastIndexOf('/>');
+        const result = icon.slice(0, index) + iterateIconProps(iconProps) + icon.slice(index);
+        return result;
+    };
```
Comment (huayunh) [inline docs/src/componentDocs/ToolbarMenu/playground/PreviewComponent.tsx:28]: These two functions look like they may be reusable utility functions. Can you place those inside the utility folder for reusability?

File: docs/src/componentDocs/ToolbarMenu/playground/PreviewComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import { ToolbarMenu } from '@brightlayer-ui/react-components/core/ToolbarMenu';
+
+import * as Colors from '@brightlayer-ui/colors';
+
+export const PreviewComponent = (): JSX.Element => {
+    const toolbarMenuJson = useAppSelector((state: RootState) => state.componentsPropsState.toolbarMenuComponent);
+
+    const toolbarMenuProps = createProps(toolbarMenuJson.props as PropsType[]);
+
+    const iterateIconProps = (iconProps: any): string => {
```
Comment (huayunh) [inline docs/src/componentDocs/ToolbarMenu/playground/PreviewComponent.tsx:16]: there are a lot of `any` types. Is there something like `IconProps` exported from MUI?

**Review by huayunh — COMMENTED** [review]
Event listeners on menu items — nice to have

File: docs/src/componentDocs/ToolbarMenu/playground/PreviewComponent.tsx (hunk 4 above)
Comment (ektasawant) [reply docs/src/componentDocs/ToolbarMenu/playground/PreviewComponent.tsx]: > there are a lot of `any` types. Is there something like `IconProps` exported from MUI?

Moved this to utlities.tsx

File: docs/src/shared/utilities.tsx [hunk 5]
```diff
@@ -155,7 +163,25 @@ export const getImage = (image: string): string => {
...
-export const filterPropsAsPerGroupType = (state: ComponentType, propName: string, groupType?: string): any => {
+const iterateIconProps = (iconProps: any): string => {
+    let str = '';
+    for (const prop in iconProps) {
+        str = `${str}` + `${prop}="${iconProps[prop]}" `;
+    }
+    return str;
+};
+
+export const getIconWithProp = (icon: string, iconProps: SvgIconProps): string => {
```
Comment (ektasawant) [inline docs/src/shared/utilities.tsx:174]: Removed any type.

**Review by huayunh — APPROVED** [review]

**Review by jeffvg — APPROVED** [review]
looks good. I see other ```iconProps: any``` was cleaned up in other component playgrounds in this PR. Do we need to have an issue logged to revisit others?

**Comment by jeffvg** [comment]
not related to this PR but noticed this today on a fresh clone. Any idea why we are seeing this?

image

**Comment by ektasawant** [comment]
> not related to this PR but noticed this today on a fresh clone. Any idea why we are seeing this?
> 
> image

Screenshot 2022-09-27 at 10 44 14 AM
This should fix the mentioned error. Will log an issue and raise PR.

File: docs/src/shared/utilities.tsx [hunk 6]
```diff
@@ -163,7 +163,25 @@ export const getImage = (image: string): string => {
     }
 };
 
-export const filterPropsAsPerGroupType = (state: ComponentType, propName: string, groupType?: string): any => {
+const iterateIconProps = (iconProps: any): string => {
```
[the remark this answers is not in the document]
Comment (ektasawant) [reply docs/src/shared/utilities.tsx:166]: If I am using `const iterateIconProps = (iconProps: SvgIconProps): string => {`
getting below error. Any suggestions?
Screenshot 2022-09-27 at 5 27 07 PM

@daileytj , @jeffvg , @huayunh , @bkarambe , @emclaug2.

**Comment by ektasawant** [comment]
> looks good. I see other `iconProps: any` was cleaned up in other component playgrounds in this PR. Do we need to have an issue logged to revisit others?

I have tested those components working fine. But still, I log an issue to test.

**Pull request merged**